kotest-tests/kotest-tests-spec-parallelism/src/jvmTest/kotlin/com/sksamuel/kotest/parallelism/ProjectConfig.kt

```diff
@@ -47,9 +49,11 @@ object ProjectConfig : AbstractProjectConfig() {
    private val TestStatusCollectorScope: CoroutineScope =
       CoroutineScope(Dispatchers.IO) + CoroutineName("TestStatusCollector")
 
-   /** Marks the start of the entire tests, when [beforeProject] is called, before any tests are launched. */
-   internal lateinit var projectStart: TimeMark
-      private set
+   /** Mark the start of the entire test case. */
+   internal val startOfTest: TimeMark = TimeSource.Monotonic.markNow()
+
+   /** Marks when [beforeProject] is called, which should be before any tests are launched. */
+   private var beforeProjectCalled: Duration? = null
```

Comment:
are any race conditions possible? Should we wrap it in `AtomicReference` or something?

Reply:
Good point, probably should be made write-only-once somehow...

Maybe make it a (threadsafe) list and ensure there's only one value.

Reply:
I added this improvement in another PR https://github.com/kotest/kotest/pull/4311